Source/WTF/wtf/text/StringImpl.h

```diff
@@ -1369,19 +1369,19 @@ inline Ref<StringImpl> StringImpl::createByReplacingInCharacters(std::span<const
 }
 
 template<typename Func>
-inline Expected<std::invoke_result_t<Func, std::span<const char>>, UTF8ConversionError> StringImpl::tryGetUTF8(const Func& function, ConversionMode mode) const
+inline Expected<std::invoke_result_t<Func, std::span<const char8_t>>, UTF8ConversionError> StringImpl::tryGetUTF8(const Func& function, ConversionMode mode) const
 {
     if (is8Bit())
         return tryGetUTF8ForCharacters(function, span8());
     return tryGetUTF8ForCharacters(function, span16(), mode);
 }
 
 template<typename Func>
-inline Expected<std::invoke_result_t<Func, std::span<const char>>, UTF8ConversionError> StringImpl::tryGetUTF8ForCharacters(const Func& function, std::span<const LChar> characters)
+inline Expected<std::invoke_result_t<Func, std::span<const char8_t>>, UTF8ConversionError> StringImpl::tryGetUTF8ForCharacters(const Func& function, std::span<const LChar> characters)
 {
     if (characters.empty()) {
-        constexpr const char* emptyString = "";
-        return function(std::span(emptyString, emptyString));
+        constexpr std::array<char8_t, 0> empty { };
+        return function(empty);
```

Comment:
The bug is that the span constructed here from `empty` ends up with `span::data()` returning nullptr. We need to write this is a way that `span::data()` returns non null even though the span is empty.

Reply:
```
    constexpr std::array<char8_t, 0> empty { };
    std::cout << "emptyData: " << empty.data() << std::endl;
```
`empty.data()` even returns nullptr in this case.

Reply:
OK. That's really annoying, but at least we understand.

Reply:
What would you think of the following then?
```
static constexpr char8_t empty = 0;
return std::span { &empty, 0 };
```

Do do you have a better proposal?

Reply:
It's not beautiful but OK I think. It can probably be done with an empty array instead of a single byte, but not sure which is better. I'd probably put that in a helper function with a name like `nonNullEmptyUTF8Span` or something a little shorter instead of repeating it three times.